Classes/Netlogix/Cqrs/Command/CommandInterface.php

```diff
@@ -31,4 +32,10 @@ public function getStatus();
 	 */
 	public function attachStatusObserver(CommandStatusObserverInterface $observer);
 
+	/**
+	 * Executes the command.
+	 *
+	 * @return void
+	 */
+	public function execute();
```

Comment:
Da bin ich dagegen. Ein asynchrones Command muss nicht wissen wie es ausgeführt wird. Das nutzen wir z.B. in der Kommunikation zwischen Traingsmgnt und unserer Webseite.

Reply:
Wir haben die ausführende Logik in den Command-Objekten implementiert, nicht in den Handlern. Das ist einer der zentralen Punkte die uns vermutlich von Flowpack etc. unterscheidet: Bei uns sind die Command-Objekte nicht nur dumme DTOs sondern enthalten die Logik.

https://github.com/netlogix/Netlogix.Cqrs.JobQueue/blob/master/Classes/Command/AsyncJobQueueCommandHandler.php#L60
Unser AsyncJobQueueCommandHandler geht davon aus, dass auch asynchrone Commands eine Execute-Methode haben. Das wird bisher nur nicht durch das Interface sichergestellt sondern knallt dann im Handler wenn die Methode nicht definiert ist.

Die Execute-Methode vom SyncInterface ins CommandInterface zu verschieben ist also nur konsistent mit der Implementierung aller Objekte.